src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -84,6 +69,29 @@ public void Move(int oldIndex, int newIndex)
         {
             MoveItem(oldIndex, newIndex);
         }
+        
+        /// <summary>
+        /// Adds a range of objects but does not raise the OnCollectionChanged event until all objects are added.
+        /// </summary>
+        /// <param name="collection">The set of objects to be added.</param>
+        /// <param name="clearFirst">Determines whether or not to remove any existing items in the collection first. Defaults to False.</param>
+        public void AddRange(IEnumerable<T> collection, bool clearFirst = false)
+        {
+            CheckReentrancy();
+
+            if (clearFirst) Items.Clear();
+
+            var newIndex = Items.Count;
+            foreach (var item in collection)
+            {
+                base.InsertItem(newIndex, item);
+                newIndex++;
+            }
```

Comment:
In reading up a bit about you, apparently you and @benaadams are "THE MEN" when it comes to perf optimizations in ,NET Core. So. looking at [Collection.cs](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/ObjectModel/Collection.cs), I'm wondering if there are any other optimizations possible from working with the list? Could just combine them all into the same PR. Thanks for the insight!